extensions/arc/src/ui/dashboards/postgres/postgresComputeAndStoragePage.ts

```diff
@@ -0,0 +1,642 @@
+		inputContainer.addItem(input, { CSSStyles: { 'margin-bottom': '10px', 'min-width': '50px', 'max-width': '225px' } });
+
+		flexContainer!.addItem(inputContainer, inputFlex);
+
+		return flexContainer;
+	}
+
+	private handleOnTextChanged(component: azdata.InputBoxComponent, currentValue: string): boolean {
+		if (component.value === currentValue) {
+			return false;
+		} else if ((!component.value) || (!component.valid)) {
+			this.discardButton!.enabled = true;
+			return false;
+		} else {
+			this.saveButton!.enabled = true;
+			this.discardButton!.enabled = true;
+			return true;
+		}
+
+	}
+
+	private editWorkerNodeCount() {
+		let currentShards = this._postgresModel.config?.spec.scale.shards;
+		let currentNodeCount = currentShards!.toString();
+
+		this.workerBox!.value = currentNodeCount;
+		this.workerBox!.min = currentShards;
+
+		this.disposables.push(
+			this.workerBox!.onTextChanged(() => {
+				if (!(this.handleOnTextChanged(this.workerBox!, currentNodeCount))) {
+					this.saveArgs!.workers = undefined;
+				} else {
+					this.saveArgs!.workers = parseInt(this.workerBox!.value!);
+				}
+			})
+		);
+	}
+
+	private vCoreMemorySection(): azdata.DivContainer {
+		const titleFlex = { flex: `0 1 250px` };
+
+		const flexContainer = this.modelView.modelBuilder.flexContainer().withLayout({
+			flexWrap: 'wrap',
+			alignItems: 'center'
+		}).component();
+
+		const titleComponent = this.modelView.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
+			value: loc.configurationPerNode,
+			CSSStyles: { ...cssStyles.title, 'font-weight': 'bold', 'margin-block-start': '0px', 'margin-block-end': '0px' }
+		}).component();
+
+		const titleContainer = this.modelView.modelBuilder.flexContainer().withLayout({ alignItems: 'center' }).component();
+		titleContainer.addItem(titleComponent, { CSSStyles: { 'margin-right': '0px', 'margin-bottom': '15px' } });
+
+		const information = this.modelView.modelBuilder.button().withProperties<azdata.ButtonProperties>({
+			iconPath: IconPathHelper.information,
+			title: loc.configurationInformation,
+			width: '12px',
+			height: '12px',
+			enabled: false
+		}).component();
+
+		titleContainer.addItem(information, { CSSStyles: { 'margin-left': '5px', 'margin-bottom': '15px' } });
+		flexContainer.addItem(titleContainer, titleFlex);
+
+		let configurationSection = this.modelView.modelBuilder.divContainer().component();
+		configurationSection.addItem(flexContainer);
+
+		return configurationSection;
+	}
+
+	private editVCores(type: string) {
+		let currentCPUSize: string | undefined;
+		if (type === loc.vCoresMax) {
+			currentCPUSize = this._postgresModel.config?.spec.scheduling?.default?.resources?.requests?.cpu;
+
+			if (!currentCPUSize) {
+				currentCPUSize = '0';
+			}
+
+			this.vCoresMaxBox!.value = currentCPUSize;
+
+			this.disposables.push(
+				this.vCoresMaxBox!.onTextChanged(() => {
```

Comment:
You're creating a listener here every time this is called? This seems to be called multiple times (such as when the discard button is pressed) so that doesn't seem correct

Reply:
Moved to initializeConfigurationBoxes() which is only called once.